src/porepy/grids/mdg_generation.py

```diff
@@ -0,0 +1,751 @@
+            mdg = pp.create_mdg("simplex", meshing_args, fracture_network)
+
+    See Also:
+        - :method:`~porepy.fracs.fracture_network_2d.FractureNetwork2d.mesh`
+        - :method:`~porepy.fracs.fracture_network_2d.FractureNetwork3d.mesh`
+        - :method:`~porepy.fracs.meshing.cart_grid`
+        - :method:`~porepy.fracs.meshing.tensor_grid`
+
+    Parameters:
+        grid_type: Type of grid. Use ``simplex`` for unstructured triangular and
+            tetrahedral grids, ``cartesian`` for structured, uniform Cartesian grids,
+            and ``tensor_grid`` for structured, non-uniform Cartesian grids.
+        meshing_args: A dictionary with meshing keys depending on each grid_type:
+            if grid_type == "simplex"
+                cell_size: ``float``: Overall minimum cell size. It is required, if one
+                    of [cell_size_min, cell_size_fracture, cell_size_boundary] is not provided.
+                cell_size_min: ``float``: minimum cell size. If not provided, cell_size
+                    will be used for completeness.
+                cell_size_fracture: ``float``: size at the fracture. If not provided,
+                    cell_size will be used for completeness.
+                cell_size_boundary: ``float``: boundary cell size. If not provided,
+                    cell_size will be used for completeness.
+            if grid_type == "cartesian"
+                cell_size: ``float``: size in any direction. It is required, if one of
+                    [cell_size_x, cell_size_y, cell_size_z] is not provided.
+                cell_size_x: ``float``: size in x-direction. If cell_size_x is provided,
+                    it overwrites cell_size in the x-direction.
+                cell_size_y: ``float``: size in y-direction. If cell_size_y is provided,
+                    it overwrites cell_size in the y-direction.
+                cell_size_z: ``float``: size in z-direction. If cell_size_z is provided,
+                    it overwrites cell_size in the z-direction.
+            if grid_type == "tensor_grid"
+                cell_size: ``float``: size in all directions. It is required, if one of
+                    [x_pts, y_pts, z_pts] is not provided.
+                x_pts: ``np.ndarray``: points in x-direction including boundary points.
+                    If x_pts is provided, it overwrites the information computed from
+                    cell_size in the x-direction.
+                y_pts: ``np.ndarray``: points in y-direction including boundary points.
+                    If y_pts is provided, it overwrites the information computed from
+                    cell_size in the y-direction.
+                z_pts: ``np.ndarray``: points in z-direction including boundary points.
+                    If z_pts is provided, it overwrites the information computed from
+                    cell_size in the z-direction.
+        fracture_network: fracture network specification.
```

Comment:
```suggestion
        fracture_network: Fracture network specification. An instance of :class:`~porepy.fracs.fracture_network_2d.FractureNetwork2d` for 2d or :class:`~porepy.fracs.fracture_network_3d.FractureNetwork3d` for 3d.
```

Reply:
Since this is already covered by type hints, I will prefer to keep it as it is now.